### DISCUSSION TITLE
I have a common SQL that cannot be parsed

### DISCUSSION THREAD

**Discussion by jiezhuCl** [JSQLParser/JSqlParser#1313] [Q&A]
I use the ccjsqlparserutil.parse method in jsqlparser to directly report an error,
Oracle Database my sql is like this:
_update taba s set (s.appname) = (select p.insuname from tableb p where p.insurednum ='3701020000008819' group by p.insuname),s.maskcode='P30'	where s.pffno =  '2671267';_

it can be used in plsql,but use this  jar will reprot error:

 **Encountered unexpected token: ")" ")"
    at line 1, column 91.
Was expecting one of:
    "&"
    ","...................**

I think it's the set a=(),b the ',' can't be explan

I use maven jar version is 4.0,
like this:
       <dependency>
            <groupId>com.github.jsqlparser</groupId>
            <artifactId>jsqlparser</artifactId>
            <version>4.0</version>
        </dependency>

**Comment by manticore-projects** [comment]
Greetings.

the problem are the extra-brackets around the column name `(s.appname)=`
Without the extra brackets your statement will be parsed:
```sql
UPDATE taba s
SET s.appname = (   SELECT p.insuname
                    FROM tableb p
                    WHERE p.insurednum = '3701020000008819'
                    GROUP BY p.insuname )
    , s.maskcode = 'P30'
WHERE s.pffno = '2671267'
;
```
I am not sure if this a bug or a feature, because the Extra-Brackets would define a `ROW` Object like in
```sql
UPDATE table_name
SET ( a, b, c) = ( 1, 2, 3 )
```

**Reply by jiezhuCl** [reply]
Do you mean that this bracket causes this problem? If it is a field query, it is easy to solve, but if it is multiple fields,
for example:
_UPDATE 
  taba s 
SET
  (
    s.appname,
    s.packagermark,
    s.packagecode
  ) = 
  (SELECT 
    p.insuname,
    p.insunvalue,
    p.billntfee 
  FROM
    tableb p 
  WHERE p.insurednum = '3701020000008819' 
  GROUP BY p.insuname),
  s.maskcode = 'P30' 
WHERE s.pffno = '2671267' ;_

 in this way, there is no way to simplify this complex SQL，Do you have any good suggestions or methods that can be corrected?

**Reply by manticore-projects** [reply]
Greetings.

your statement as a syntax error `s.maskcode = 'P30'` but will work otherwise:

```sql
UPDATE taba s
SET (   s.appname
        , s.packagermark
        , s.packagecode ) = (   SELECT  p.insuname
                                        , p.insunvalue
                                        , p.billntfee
                                FROM tableb p
                                WHERE p.insurednum = '3701020000008819'
                                GROUP BY p.insuname )
WHERE s.pffno = '2671267'
;
```
You can [test it online here](http://jsqlformatter.manticore-projects.com/jsqlformatter/demo.html?args=-c%20KoBQIgggKgogBAFwIYCMlwM4CgDKMpwAUcJGAdEgA6UB2SAtgKZYmusA0mZlSAxgNZIA5owBO9JKP4s2HLjwHDGvAPYATRnACUcALxFWeADIwAwgTiUyASxoYArnSYzZrt+49yrthzQBuSAA29syeYeFunFYo1oGBNAgAZoyhEWlsAGIASgDyALKIqIGMKJYu6WkA6gASMFnw3nb2ooxqNPb0enAA5ADMAOwADACMgwBMg5NTABzTwwCc3eUV4QDiuaBwAEIAmpY2TU6aWlg1dfDklImJNCpd3WMAbP3DT-1LANxAA+)

**Reply by jiezhuCl** [reply]
I know what you mean. I use the website provided by you to verify. Indeed, this SQL will report errors, but I use it in oralcle PL / SQL developer. It is normal and can be executed and submitted：
_UPDATE 
  prpjpaymentbill b 
SET
  (
    b.packagecode,
    b.packageremark,
    b.agentcode
  ) = 
  (SELECT 
    p.payrefreason,
    p.classcode,
    p.riskcode 
  FROM
    prpjcommbill p where p.policertiid='SDDH200937010330006366' ),
  b.payrefnotype = '05',
  b.packageunit = '4101170402'
  where b.payrefno='B370202091026000005' ;
_
![图片](https://user-images.githubusercontent.com/33033276/131058278-7e22b744-d550-4452-b3a0-15e1b94f1cd1.png)

![图片](https://user-images.githubusercontent.com/33033276/131058398-964dd719-59ae-4889-972d-dd2cf8540ce3.png)

**Comment by manticore-projects** [comment]
Details as below, all these samples work on H2 and I will look into it.
I opened issue #1316 accordingly.

```sql
CREATE TABLE test (
    a        VARCHAR (1)
    , b      VARCHAR (1)
)
;
-- queries are supported
update test
set (a, b) = (select '1', '2');

-- value lists are not supported
update test
set (a, b) = values( '1', '2');

-- simple expression lists are not supported
update test
set (a, b) = ('1', '2');

-- complex expression lists are not supported
update test
set (a, b) = (1, (select 2));
```

**Reply by jiezhuCl** [reply]
OK, I see. Thank you very much！

**Reply by manticore-projects** [reply]
PR #1317 solves this issue. Thanks for reporting.

**Reply by jiezhuCl** [reply]
I have verified that it is indeed OK. Thank you again. Thank you very much!
